packages/core/contracts-ovm/insured-bridge/implementation/OVM_BridgeDepositBox.sol

```diff
@@ -1,47 +1,237 @@
+        depositsEnabled = _depositsEnabled;
+        emit DepositsEnabled(_depositsEnabled);
+    }
 
+    /**************************************
+     *         DEPOSITOR FUNCTIONS        *
+     **************************************/
+
+    /**
+     * @notice Called by L2 user to bridge funds between L2 and L1.
+     * @dev Emits the `L2FundsDeposited` event which relayers listen for as part of the bridging action.
+     * @dev The caller must first approve this contract to spend `amount` of `l2Token`.
+     * @param recipient L1 address that should receive the tokens.
+     * @param l2Token L2 token to deposit.
+     * @param amount How many L2 tokens should be deposited.
+     * @param maxFee Max fraction of the total `amount` that the depositor is willing to pay as a fee. scaled by 1e18.
+     */
     function deposit(
         address recipient,
-        address l1Token,
+        address l2Token,
         uint256 amount,
         uint256 maxFee
-    ) public {}
+    ) public onlyIfDepositsEnabled() {
+        require(isTokenWhitelisted(l2Token), "deposit token not whitelisted");
+        require(maxFee <= 1e18, "max fee can not be over 1e18");
+
+        TokenHelper.safeTransferFrom(l2Token, msg.sender, address(this), amount);
+
+        emit L2FundsDeposited(
+            numberOfDeposits, // the current number of deposits acts as a deposit ID (nonce).
+            getCurrentTime(),
+            msg.sender,
+            recipient,
+            whitelistedTokens[l2Token].l1Token,
+            amount,
+            maxFee
+        );
+
+        numberOfDeposits += 1;
+    }
+
+    /**************************************
+     *          RELAYER FUNCTIONS         *
+     **************************************/
+
+    /**
+     * @notice Called by relayer (or any other EOA) to move a batch of funds from the deposit box, through the canonical
+     *      token bridge, to the L1 Withdraw box.
+     * @dev The frequency that this function can be called is rate limited by the `minimumBridgingDelay` to prevent spam
+     *      on L1 as the finalization of a L2->L1 tx is quite expensive.
+     * @param l2Token L2 token to relay over the canonical bridge.
+     * @param l1Gas Unused by optimism, but included for potential forward compatibility considerations.
+     */
+    function bridgeTokens(address l2Token, uint32 l1Gas) public {
+        uint256 bridgeDepositBoxBallance = TokenLike(l2Token).balanceOf(address(this));
+        require(bridgeDepositBoxBallance > 0, "can't bridge zero tokens");
+        require(isTokenWhitelisted(l2Token), "can't bridge non-whitelisted token");
+        require(hasEnoughTimeElapsedToBridge(l2Token), "not enough time has elapsed from previous bridge");
+
+        StandardBridgeLike(Lib_PredeployAddresses.L2_STANDARD_BRIDGE).withdrawTo(
+            l2Token, // _l2Token. Address of the L2 token to bridge over.
+            l1WithdrawContract, // _to. Withdraw, over the bridge, to the l1Withdraw contract.
```

Comment:
Do we need to check that the withdrawContract is not 0x0? Or do we do this elsewhere? Just want to make sure we're not ever able to send tokens to the 0x0 address

Reply:
hhhmmm. that is an interesting point. we could add a modifier that checks `isWithdrawContractSet` or a `require` for this. wdyt?

Reply:
I think use a modifier. I plan to add a modifier for checking whether the depositContract is set and using that on any functions that call `sendCrossDomainMessage`

Reply:
what I did to combat this situation is add a require check in the initialization of this state variable to ensure it is set correctly so we can never get into a state where the withdrawContract is 0x0.